packages/ember-routing/lib/location/none_location.js

```diff
@@ -23,14 +23,30 @@ export default EmberObject.extend({
   path: '',
 
   /**
-    Returns the current path.
+    Will be pre-pended to path.
+
+    @private
+    @property rootURL
+    @default '/'
+  */
+  rootURL: '/',
+
+  /**
+    Returns the current path without `rootURL`.
 
     @private
     @method getURL
     @return {String} path
   */
   getURL() {
-    return get(this, 'path');
+    let path = get(this, 'path');
+    let rootURL = get(this, 'rootURL');
+
+    // remove trailing slashes if they exists
+    rootURL = rootURL.replace(/\/$/, '');
+
+    // remove rootURL from url
+    return path.replace(rootURL, '');
```

Comment:
Given a `rootURL` of `/foo` we want to strip it from `/foo/bar/baz` but we do not want `/foo/baz/foo/bar` to become `/baz/bar`.